lib/spack/spack/cmd/list.py

```diff
@@ -306,6 +308,11 @@ def list(parser, args):
     # Filter the set appropriately
     sorted_packages = filter_by_name(pkgs, args)
 
+    # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        pkgs = [p for p in pkgs if p in packages_with_tags]
```

Comment:
You want to be working with the packages filtered by name at this point *and* make sure you are passing the filtered results to the next filter option.

```suggestion
        sorted_packages = [p for p in sorted_packages if p in packages_with_tags]
```